File: Ej1/src/practica1ds/FactoriaCarretera.java
package practica1ds;

public class FactoriaCarretera implements FactoriaCarreraYBicicleta {

    @Override
    public Carrera crearCarrera() {
        CarreraCarretera carrera = new CarreraCarretera("Carretera");
        return carrera;
    }

    @Override
    public Bicicleta crearBicicleta(String id) {
        BicicletaCarretera bicicleta = new BicicletaCarretera("Carretera", id);
        return bicicleta;
    }

}

File: Ej2/src/factoria_carretera.py
from src.impl.bicicleta_carretera import BicicletaCarretera
from src.impl.carrera_carretera import CarreraCarretera


class FactoriaCarretera:
    def __init__(self):
        self.num_bicicletas = 0

    def crear_carrera(self):
        return CarreraCarretera()

    def crear_bicicleta(self):
        identificador = self.num_bicicletas
        self.num_bicicletas += 1
        return BicicletaCarretera(identificador=identificador)
